### PR TITLE
Configurable API & Dashboard base path

### LINKED ISSUE #5853: Allow to add path preifx for both api and dashboard endpoints
### Do you want to request a *feature* or report a *bug*?

Feature

### What did you expect to see?

As #5374 has been fixed, now dashboard assets can be served under any path prefix, but it continues to fetch api data from `/api`, which seems to be determined at build time or hard coded (?)

I would suggest to add something like `--api.pathprefix=/path/prefix` then these endpoints will be served under `/path/prefix/api` and `/path/prefix/dashboard`. Which will make them easier to integrate into existing auth system: specific cookie under some path prefix.

### PR DESCRIPTION
**Pull request by rtribotte**
### What does this PR do?

This PR adds an option to configure the base path where the API and dashboard will be exposed.

### Motivation

Fixes #5853 

### More

- [ ] Added/updated tests
- [x] Added/updated documentation

### Additional Notes

Co-authored-by: Kevin Pollet <pollet.kevin@gmail.com>

### CODE REVIEW

File: docs/content/operations/dashboard.md [hunk 1]
```diff
@@ -104,8 +104,44 @@ rule = "Host(`traefik.example.com`) && (PathPrefix(`/api`) || PathPrefix(`/dashb
 ??? example "Dashboard Dynamic Configuration Examples"
     --8<-- "content/operations/include-dashboard-examples.md"
 
+### Custom API Base Path
+
+As shown above, by default Traefik exposes its API and Dashboard under the `/` base path,
+which means that respectively the API is served under the `/api` path,
+and the dashboard under the `/dashboard` path.
+
+However, it is possible to configure this base path:
```
Comment (mloiseleur) [inline docs/content/operations/dashboard.md:113]:
```suggestion
It is possible to configure this base path:
```

File: pkg/api/dashboard/dashboard.go [hunk 2]
```diff
@@ -39,24 +92,28 @@ func Append(router *mux.Router, customAssets fs.FS) {
...
-	if assets == nil {
-		assets = webui.FS
-	}
-	// allow iframes from our domains only
-	// https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/frame-src
-	w.Header().Set("Content-Security-Policy", "frame-src 'self' https://traefik.io https://*.traefik.io;")
+	router.Methods(http.MethodGet).
+		PathPrefix(dashboardPath).
+		HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			// allow iframes from our domains only
```
Comment (mloiseleur) [inline pkg/api/dashboard/dashboard.go:106]:
```suggestion
			// allow iframes from traefik domains only
```

**Review by lbenguigui — APPROVED** [review]
LGTM

File: docs/content/operations/dashboard.md (hunk 1 above)
Comment (rtribotte) [reply docs/content/operations/dashboard.md:113]: @mloiseleur I'm not sure about this suggestion, why suppressing this adverb?

File: pkg/api/dashboard/dashboard.go [hunk 3]
```diff
@@ -1,35 +1,88 @@
...
+
+	// allow iframes from traefik domains only
+	// https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/frame-src
+	w.Header().Set("Content-Security-Policy", "frame-src 'self' https://traefik.io https://*.traefik.io;")
+
+	// The content type must be guessed by the file server.
+	// https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/X-Content-Type-Options
+	w.Header().Del("Content-Type")
+
+	if r.URL.Path == "/" {
```
Comment (jspdown) [inline pkg/api/dashboard/dashboard.go:41]: When testing using a `--api.insecure` I don't fall into this condition and end up with a non-templated `index.html`.

**Review by jspdown — CHANGES_REQUESTED** [review]

**Review by jspdown — APPROVED** [review]
LGTM :+1:

**Pull request merged**